### DISCUSSION TITLE
Add support for Web Components and Shadow DOM ?

### DISCUSSION THREAD

**Discussion by 0xJohnnyboy** [tailwindlabs/tailwindcss#7217] [Ideas]
I'm currently exploring [Lit](https://lit.dev/) which is a JS/TS framework based on web components. It works well but as it uses shadow DOM, I couldn't figure out a way to make tailwind work except for some obscure anti-patterns workarounds.

Would someone consider working on some plugin to add support for shadow DOM ? That could be a tool in tailwindcss-cli to use at build time or whatever... I was thinking of using Lit's static styles abilities by having ts constant files but that doesn't seem like a great solution.

I don't have the time nor the skills to do something like this but this would be super interesting as that kind of framework is growing. This would insure that tailwindcss stays future-proof and integrates with new web standards.

Plus I would be so disappointed never to be able to use Tailwindcss properly with web components. Especially with that third version 🔥 

👋

**Comment by shaynem** [comment]
Came across a similar issue 

have a look at these solution

[Build time](https://dev.to/43081j/using-tailwind-at-build-time-with-web-components-1bhm)

or 

[Run time](https://dev.to/43081j/using-tailwind-at-run-time-with-web-components-47c)
